File: apps/packages/flutter_app_core/lib/src/shared/models/agent_role_models.dart
class AgentRoleInfo {
  const AgentRoleInfo({
    required this.id,
    required this.code,
    required this.name,
    required this.displayName,
    required this.description,
    required this.status,
    required this.icon,
    required this.coverImage,
    required this.currentReleaseId,
    required this.currentVersion,
    required this.publishedAt,
    required this.sort,
  });

  final String? id;
  final String? code;
  final String? name;
  final String? displayName;
  final String? description;
  final String? status;
  final String? icon;
  final String? coverImage;
  final String? currentReleaseId;
  final String? currentVersion;
  final String? publishedAt;
  final int sort;

  String get title {
    final value = displayName ?? name ?? code ?? '未命名智能体角色';
    return value.trim().isEmpty ? '未命名智能体角色' : value;
  }

  factory AgentRoleInfo.fromJson(Map<String, dynamic> json) {
    return AgentRoleInfo(
      id: json['id']?.toString(),
      code: json['code']?.toString(),
      name: json['name']?.toString(),
      displayName: json['displayName']?.toString(),
      description: json['description']?.toString(),
      status: json['status']?.toString(),
      icon: json['icon']?.toString(),
      coverImage: json['coverImage']?.toString(),
      currentReleaseId: json['currentReleaseId']?.toString(),
      currentVersion: json['currentVersion']?.toString(),
      publishedAt: json['publishedAt']?.toString(),
      sort: (json['sort'] as num?)?.toInt() ?? 0,
    );
  }
}

class AgentRoleReleaseInfo {
  const AgentRoleReleaseInfo({
    required this.id,
    required this.roleId,
    required this.version,
    required this.status,
    required this.publishedAt,
  });

  final String? id;
  final String? roleId;
  final String? version;
  final String? status;
  final String? publishedAt;

  factory AgentRoleReleaseInfo.fromJson(Map<String, dynamic> json) {
    return AgentRoleReleaseInfo(
      id: json['id']?.toString(),
      roleId: json['roleId']?.toString(),
      version: json['version']?.toString(),
      status: json['status']?.toString(),
      publishedAt: json['publishedAt']?.toString(),
    );
  }
}

File: admin-web/src/app/features/agent-roles/shared/agent-role.models.ts
export interface AgentRolePluginMappingItem {
  pluginId?: string | null;
  paramInfo?: string | null;
  providerCode?: string | null;
}

export interface AgentRoleContextProviderItem {
  id?: string | null;
  url?: string | null;
  headers?: string | null;
}

export interface AgentRoleKnowledgeBindingItem {
  id?: number | null;
  datasetId?: string | null;
  datasetName?: string | null;
  sort?: number | null;
  enabled?: boolean | null;
  topK?: number | null;
  minScore?: number | null;
  rerankEnabled?: boolean | null;
  maxChunks?: number | null;
}

export interface AgentRoleItem {
  id?: string | null;
  code?: string | null;
  name?: string | null;
  displayName?: string | null;
  description?: string | null;
  status?: string | null;
  isSystem?: boolean;
  icon?: string | null;
  coverImage?: string | null;
  themeToken?: string | null;
  asrProfileId?: string | null;
  vadProfileId?: string | null;
  llmProfileId?: string | null;
  ttsProfileId?: string | null;
  ttsVoiceId?: string | null;
  ttsLanguage?: string | null;
  ttsVolume?: number | null;
  ttsRate?: number | null;
  ttsPitch?: number | null;
  chatHistoryConf?: number | null;
  systemPrompt?: string | null;
  knowledgeNoHitFallbackText?: string | null;
  memoryEnabled?: boolean;
  memoryModelId?: string | null;
  memoryEntityScope?: string | null;
  memoryTopK?: number | null;
  memoryMinScore?: number | null;
  memoryIntentRecall?: boolean | null;
  memoryQueryRewrite?: boolean | null;
  memoryRerank?: boolean | null;
  langCode?: string | null;
  language?: string | null;
  currentReleaseId?: string | null;
  currentVersion?: string | null;
  publishedAt?: string | null;
  sort?: number;
  creator?: number | null;
  createdAt?: string | null;
  updater?: number | null;
  updatedAt?: string | null;
  pluginMappings?: AgentRolePluginMappingItem[];
  contextProviders?: AgentRoleContextProviderItem[];
  knowledgeBindings?: AgentRoleKnowledgeBindingItem[];
}

export interface AgentRoleBindingOptions {
  asrProfiles: { label: string; value: string; description?: string | null }[];
  vadProfiles: { label: string; value: string; description?: string | null }[];
  llmProfiles: { label: string; value: string; description?: string | null }[];
  ttsProfiles: { label: string; value: string; description?: string | null }[];
}

export interface AgentRoleReleaseItem {
  id?: string | null;
  roleId?: string | null;
  version?: string | null;
  status?: string | null;
  snapshotJson?: string | null;
  remark?: string | null;
  publishedAt?: string | null;
  creator?: number | null;
  createdAt?: string | null;
  updater?: number | null;
  updatedAt?: string | null;
}

export interface AgentRolePageData {
  total: number;
  list: AgentRoleItem[];
}

export interface AgentRolePayload {
  code: string;
  name: string;
  displayName?: string;
  description?: string;
  status?: string;
  icon?: string;
  coverImage?: string;
  themeToken?: string;
  asrProfileId?: string;
  vadProfileId?: string;
  llmProfileId?: string;
  ttsProfileId?: string;
  ttsVoiceId?: string;
  ttsLanguage?: string;
  ttsVolume?: number;
  ttsRate?: number;
  ttsPitch?: number;
  chatHistoryConf?: number;
  systemPrompt?: string;
  knowledgeNoHitFallbackText?: string;
  memoryEnabled?: boolean;
  memoryModelId?: string;
  memoryEntityScope?: string;
  memoryTopK?: number;
  memoryMinScore?: number;
  memoryIntentRecall?: boolean;
  memoryQueryRewrite?: boolean;
  memoryRerank?: boolean;
  langCode?: string;
  language?: string;
  sort?: number;
  pluginMappings?: AgentRolePluginMappingItem[];
  contextProviders?: AgentRoleContextProviderItem[];
  knowledgeBindings?: AgentRoleKnowledgeBindingItem[];
}
